include/evse_security/detail/openssl/openssl_types.hpp

```diff
@@ -33,6 +33,13 @@ template <> class std::default_delete<X509_REQ> {
     }
 };
 
+template <> class std::default_delete<STACK_OF(X509)> {
+public:
+    void operator()(STACK_OF(X509) * ptr) const {
+        sk_X509_free(ptr);
```

Comment:
it might need to be sk_X509_pop_free() rather than sk_X509_free().
In newer versions of OpenSSL (3.2) there is OSSL_STACK_OF_X509_free().
The implementation uses sk_X509_pop_free().

Reply:
I assume that valgrind/memcheck is happy.